### ISSUE TITLE
Please change the default encoding to UTF-8

### ISSUE THREAD

**Issue by ashraf21c** [ampache/ampache#3763]
Hello,
Arabic isn`t ccorrectly  displayed unless user changes the encoding to UTF-8.
As far as I know, it`s the default now for websites using UTF-8.
However, this script isn`t.
Edge doesn`t have the encoding setting thats was available in IE.
Firefox requires as addon as an easy solution for the normal user.

Kindly force UTF-8 encoding in Ampache script.

**Comment by lachlan-00** [comment]
the database default is utf8mb4
site_charset defaults to UTF-8

which parts are not in utf?

**Comment by ashraf21c** [comment]
All Arabic text in the backend (admin dashboard) isn`t readable unless I use an set it to UTF-8 using an addon.
In addition, it requires that manual UTF-8 settings to each page. This means when I navigate from a tab/ page to another (yes, even tabs in the same page), I have to click the addon icon and select UTF-8 charset, which is very frustrating.

**Comment by ashraf21c** [comment]
I shall test UTF-8 variations in the database.

> the database default is utf8mb4

**Comment by ashraf21c** [comment]
Update:

(1) Changing database encoding didn`t work.
(2) I tried forcing the encoding in the PHP file by adding the encoding tag in the header of "index.php" and "preferences.php", just for testing.. 

`<?php
header('Content-Type: text/html; charset=utf-8');`

However, this didn`t work either.
In fact, just adding this line caused error 500.
I think the "text/html" needs to be changed. (I left PHP many years ago and use only python now, so, I forgot nearly everything concerning PHP).

How about releasing an update with the correct tag inserted in all php files?

In HTML, I always force both UTF-8 and RTL using this code:

`<!DOCTYPE html>
<html dir="rtl" lang="ar">
<head>
<meta charset="utf-8">`

Here, I suggest applying the same scenario in PHP files.

**Comment by ashraf21c** [comment]
> site_charset defaults to UTF-8

I checked the charset of "index.html" while it`s opened in the browser, with the help of an addon.. It shows that the charset in use is "Arabic (Windows 1256)".
When I change it to "UTF-8" it works.
So, the default charset in AMpache is "Arabic (Windows 1256)", not "UTF-8". It needs to be forced.
I used Microsoft Edge and Firefox in two PCs.. I got the same result.

**Comment by mitchray** [comment]
I'm seeing it as UTF-8 in the source

```
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en-US" lang="en-US" dir="ltr">
    <head>
        <!-- Propelled by Ampache | ampache.org -->
        <link rel="search" type="application/opensearchdescription+xml" title="Ampache :: For the Love of Music" href="[http://ampache-preview/search.php?action=descriptor](view-source:http://ampache-preview/search.php?action=descriptor)" />
                <meta http-equiv="Content-Type" content="application/xhtml+xml; charset=UTF-8" />
```

**Comment by ashraf21c** [comment]
> I'm seeing it as UTF-8 in the source
> 
> ```
> <!DOCTYPE html>
> <html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en-US" lang="en-US" dir="ltr">
>     <head>
>         
>         <link rel="search" type="application/opensearchdescription+xml" title="Ampache :: For the Love of Music" href="[http://ampache-preview/search.php?action=descriptor](view-source:http://ampache-preview/search.php?action=descriptor)" />
>                 <meta http-equiv="Content-Type" content="application/xhtml+xml; charset=UTF-8" />
> ```

Thank you.
I don`t know why this happens.
Is there a chance for the charset being changed when applying the Arabic UI?

**Comment by ashraf21c** [comment]
> site_charset

It is blank

**Comment by mitchray** [comment]
In the Admin tab > Server Config > Ampache Debug page, what does it say is "site_charset"

**Comment by ashraf21c** [comment]
> In the Admin tab > Server Config > Ampache Debug page, what does it say is "site_charset"

It is blank.

![site charset](https://github.com/ampache/ampache/assets/38018061/c8589132-4948-4937-b45a-615533dd4851)

**Comment by mitchray** [comment]
That is strange, set it to the below in your config/ampache.cfg.php as I'm guessing its blank/missing there somehow?

```site_charset = "UTF-8"```

**Comment by ashraf21c** [comment]
> That is strange, set it to the below in your config/ampache.cfg.php as I'm guessing its blank/missing there somehow?
> 
> `site_charset = "UTF-8"`

Thank you very much! The problem is fixed.

This is the screenshot before editing the line.
![charset- config](https://github.com/ampache/ampache/assets/38018061/b35022b4-4f50-4421-9063-313cb9738cb7)

I guess that the charset gets changed when changing the language. (A possible bug??)
Another possible matter to consider is that this is a Softaculous installation).

To replicate this problem, just change the UI to Arabic.

**Comment by ashraf21c** [comment]
Update:
I decided to help checking this.
I removed the current installation and re-installed it.
I found that, as a fresh installation, before even logging in for the first time, mentioned line is empty.
Even when selecting English as installation language.
So, this problem applies only to Softaculous installation. They use version 6.2.1 (Dated: 09/01/2024). I`m sure that their installer has some misbehaviour, as the GitHub copy has the full line included.

Well, this is a bug to report to Softaulous. I`ll report it to them.
Sorry for bothering you, and many thanks for helping me.
